### PR TITLE
Refactor script to write data to SQL database

### PR DESCRIPTION
**Pull request by fhallee**

### CODE REVIEW

File: citylex.py [hunk 1]
```diff
@@ -358,9 +335,10 @@ def _udlexicons(lexicon: citylex_pb2.Lexicon) -> None:
...
 
-def _unimorph(lexicon: citylex_pb2.Lexicon) -> None:
-    """Collects UniMorph analyses."""
+def _unimorph(conn: str) -> None:
+    """Collects UniMorph analyses and inserts them into the database."""
     counter = 0
+    cursor = conn.cursor()
     url = "https://raw.githubusercontent.com/unimorph/eng/master/eng"
     for line in _request_url_resource(url):
         line = line.rstrip()
```
Comment (kylebgorman) [inline citylex.py:344]: I don't remember why I did it this way but consider using a `csv.reader` or a `csv.DictReader` here instead. It's probably safer.

File: citylex.py [hunk 2]
```diff
@@ -381,17 +356,18 @@ def _unimorph(lexicon: citylex_pb2.Lexicon) -> None:
 # WikiPron-UK.
 
 
-def _wikipron_uk(lexicon: citylex_pb2.Lexicon) -> None:
-    """Collects WikiPron US pronunciations."""
+def _wikipron_uk(conn: str) -> None:
```
Comment (kylebgorman) [inline citylex.py:359]: connection object is not a string, contra typing

File: citylex.py [hunk 3]
```diff
@@ -381,17 +356,18 @@ def _unimorph(lexicon: citylex_pb2.Lexicon) -> None:
...
+    """Collects WikiPron UK pronunciations and inserts them into the database."""
     counter = 0
+    cursor = conn.cursor()
     url = (
-        "https://raw.githubusercontent.com/cuny-cl/"
+        "https://raw.githubusercontent.com/kylebgorman/"
         "wikipron/master/data/scrape/tsv/eng_latn_uk_broad_filtered.tsv"
     )
     for line in _request_url_resource(url):
```
Comment (kylebgorman) [inline citylex.py:367]: ditto; I don't know why this isn't a `csv` reader or something.

File: citylex.py [hunk 4]
```diff
@@ -400,37 +376,32 @@ def _wikipron_uk(lexicon: citylex_pb2.Lexicon) -> None:
 # WikiPron-US.
 
 
-def _wikipron_us(lexicon: citylex_pb2.Lexicon) -> None:
-    """Collects WikiPron US pronunciations."""
+def _wikipron_us(conn: str) -> None:
```
Comment (kylebgorman) [inline citylex.py:379]: connection object is not a string, contra typing

File: citylex.py [hunk 5]
```diff
@@ -400,37 +376,32 @@ def _wikipron_uk(lexicon: citylex_pb2.Lexicon) -> None:
...
+    """Collects WikiPron US pronunciations and inserts them into the database."""
     counter = 0
+    cursor = conn.cursor()
     url = (
-        "https://raw.githubusercontent.com/cuny-cl/"
+        "https://raw.githubusercontent.com/kylebgorman/"
         "wikipron/master/data/scrape/tsv/eng_latn_us_broad_filtered.tsv"
     )
     for line in _request_url_resource(url):
```
Comment (kylebgorman) [inline citylex.py:387]: ditto, I don't know why this isn't a CSV reader

File: citylex.py [hunk 6]
```diff
@@ -400,37 +376,32 @@ def _wikipron_uk(lexicon: citylex_pb2.Lexicon) -> None:
...
         (wordform, pron, *_) = line.rstrip().split("\t")
         wordform = _normalize(wordform)
-        lexicon.entry[wordform].wikipron_us_pron.append(pron)
+        cursor.execute("INSERT INTO pronunciation (wordform, dialect, source, standard, pronunciation, is_observed) VALUES (?, ?, ?, ?, ?, ?)", (wordform, "US", "WikiPron US", "IPA", pron, 1))
```
Comment (kylebgorman) [inline citylex.py:390]: use `True` instead of `1`

File: citylex.py [hunk 7]
```diff
@@ -218,70 +196,75 @@ def _elp(lexicon: citylex_pb2.Lexicon) -> None:
...
     path = "SUBTLEX-US frequency list with PoS information text version.txt"
-    source = _request_url_zip_resource(url, path)
-    for drow in csv.DictReader(source, delimiter="\t"):
-        wordform = _normalize(drow["Word"])
-        ptr = lexicon.entry[wordform]
-        ptr.subtlex_us_freq = int(drow["FREQcount"])
-        ptr.subtlex_us_cd = int(drow["CDcount"])
+    file = _request_url_zip_resource(url, path)
+    cursor = conn.cursor()
+    total_freq = 0
```
Comment (kylebgorman) [inline citylex.py:256]: ditto previous comment; also you can do this in SQLite now

File: citylex.py [hunk 8]
```diff
@@ -381,17 +356,18 @@ def _unimorph(lexicon: citylex_pb2.Lexicon) -> None:
...
     url = (
-        "https://raw.githubusercontent.com/cuny-cl/"
+        "https://raw.githubusercontent.com/kylebgorman/"
         "wikipron/master/data/scrape/tsv/eng_latn_uk_broad_filtered.tsv"
     )
     for line in _request_url_resource(url):
         (wordform, pron, *_) = line.rstrip().split("\t")
         wordform = _normalize(wordform)
-        lexicon.entry[wordform].wikipron_uk_pron.append(pron)
+        cursor.execute("INSERT INTO pronunciation (wordform, dialect, source, standard, pronunciation, is_observed) VALUES (?, ?, ?, ?, ?, ?)", (wordform, "UK", "WikiPron UK", "IPA", pron, 1))
```
Comment (kylebgorman) [inline citylex.py:370]: use `True` instead of `1`

File: citylex.py [hunk 9]
```diff
@@ -494,79 +459,79 @@ def main() -> None:
     )
     args = parser.parse_args()
 
-    # Builds TSV fieldnames and lexicon.
-    lexicon = citylex_pb2.Lexicon()
-    fieldnames = []
+    conn = sqlite3.connect(args.db_path)
+    cursor = conn.cursor()
+
```
Comment (kylebgorman) [inline citylex.py:464]: Pet peeve of mine but I can't stand blank lines inside of a function. Use comments or something if you need to space it out, or break it apart into smaller functions.

File: citylex.py [hunk 10]
```diff
@@ -494,79 +459,79 @@ def main() -> None:
...
+    logging.info("Dropping existing tables if they exist...")
+    cursor.execute("DROP TABLE IF EXISTS frequency;")
```
Comment (kylebgorman) [inline citylex.py:466]: Why not use a loop here? Also note that `;` is never functional in Python's `sqlite3`

**Review by kylebgorman — COMMENTED** [review]
Thanks for the submission, this looks really good. Please:

* Address issues below; you can click "resolve" once you've `push`ed a version that addresses them to let me know you've taken care of it. A few may not be easily soluble; if so we can chat on email. 
* Run the third-party tool `black` over the file: `black -l79 citylex.py`. This should fix all line wraps and overlong lines except for comments (which you have to wrap manually).
* Run the third-party tool `flake8` over the file: `flake8 citylex.py`. Fix any errors it records.

I think then we're ready to go. Thanks again.

**Review by kylebgorman — COMMENTED** [review]
This looks great, thanks. I have just a few last things before we finalize this: 

* Right now it's set up to commit the changes at the very end. Since each step is writing to a different database, I'd probably commit changes at the end of each of the major functions instead.  There's no reason to wait until the very end.
* You can also remove the commit at the very end.
* Please remove the protobuf dependency line from these two files:

https://github.com/CUNY-CL/citylex/blob/master/pyproject.toml
https://github.com/CUNY-CL/citylex/blob/master/requirements.txt

(I obviously can do these myself but we usually let the submitter take care of tehse details.)

**Comment by fhallee** [comment]
Done!

**Review by kylebgorman — APPROVED** [review]
LGTM. (I'm going to run the tests locally before I merge just to be sure.)

**Pull request merged**